tensorflow/core/kernels/sparse_tensor_dense_matmul_op_gpu.cu.cc

```diff
@@ -140,69 +192,17 @@ struct SparseTensorDenseMatMulFunctor<GPUDevice, T, Tindices, ADJ_A, ADJ_B> {
             "available for data of type tf.float64 or tf.complex128");
       }
       using Tsum = typename SumType<T>::type_for_determinism;
-      ComputeWithPossiblyHigherPrecisionAccumulator<Tsum>(
+      ComputeWithPossiblyHigherPrecisionAccumulator<Tsum, T, Tindices,
+                                                    ADJ_A, ADJ_B>(
           ctx, out, a_indices, a_values, b);
     } else {
       using Tsum = typename SumType<T>::type;
-      ComputeWithPossiblyHigherPrecisionAccumulator<Tsum>(
+      ComputeWithPossiblyHigherPrecisionAccumulator<Tsum, T, Tindices,
+                                                    ADJ_A, ADJ_B>(
           ctx, out, a_indices, a_values, b);
     }
     return Status::OK();
   }
-
-  private:
-    template <typename Tsum>
-    static EIGEN_ALWAYS_INLINE Status
-    ComputeWithPossiblyHigherPrecisionAccumulator(
```

Comment:
@wenscarl, can you explain why making this template function a private static method of the functor doesn't work for the ROCm build? Does it build for you locally?